webapp/src/main/java/life/qbic/views/login/LoginLayout.java

```diff
@@ -33,163 +25,81 @@
-    passwordTooShortMessage.setVisible(false);
-    emailConfirmationFailedMessage = new ErrorMessage("Unconfirmed email address",
-        "Please confirm your email address before logging in");
-    emailConfirmationFailedMessage.setVisible(false);
-  }
-
-  private void createInformationDivs() {
-    confirmationInformationMessage = new InformationMessage("Email address confirmed",
-        "You can now login with your credentials.");
-    confirmationInformationMessage.setVisible(false);
-  }
-
   private void styleFormLayout() {
-    contentLayout.addClassNames(
-        "bg-base",
-        "border",
-        "border-contrast-10",
-        "rounded-m",
-        "box-border",
-        "flex",
-        "flex-col",
-        "w-full",
-        "text-s",
-        "shadow-l",
-        "min-width-300px",
-        "max-width-15vw");
-    contentLayout.add(
-        layoutTitle,
-        errorMessage,
-        confirmationInformationMessage,
-        email,
-        password,
-        loginButton,
-        forgotPasswordButton,
-        registerSpan);
-  }
-
-  private void createEmailField() {
-    email = new EmailField("Email");
-  }
-
-  private void createPasswordField() {
-    password = new PasswordField("Password");
-  }
-
-  private void createLoginButton() {
-    loginButton = new Button("Log In");
-  }
-
-  private void styleLoginButton() {
-    loginButton.setWidthFull();
-    loginButton.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
-  }
-
-  private void createForgotPasswordButton() {
-    forgotPasswordButton = new Button("Forgot Password");
-  }
-
-  private void styleForgotPasswordButton() {
-    forgotPasswordButton.setWidthFull();
-    forgotPasswordButton.addThemeVariants(ButtonVariant.LUMO_TERTIARY_INLINE);
-  }
-
-  private void createRegisterSpan() {
+    registerSpan.addClassName("p-l");
+    contentLayout.setWidthFull();
+    contentLayout.setPadding(false);
+    contentLayout
+        .getElement()
+        .getClassList()
+        .addAll(
+            List.of(
+                "bg-base",
+                "border",
+                "border-contrast-10",
+                "rounded-m",
+                "box-border",
+                "flex",
+                "flex-col",
+                "w-full",
+                "text-s",
+                "shadow-l",
+                "min-width-300px",
+                "max-width-15vw"));
+  }
+
+  private void createSpan() {
     RouterLink routerLink = new RouterLink("REGISTER", UserRegistrationLayout.class);
     registerSpan = new Span(new Text("Need an account? "), routerLink);
   }
 
-  private void setRequiredIndicatorVisible(HasValueAndElement<?, ?>... components) {
-    Stream.of(components).forEach(comp -> comp.setRequiredIndicatorVisible(true));
-  }
   @Override
-  public void setParameter(BeforeEvent beforeEvent, @OptionalParameter String s) {
-    loginHandlerInterface.handle(beforeEvent);
+  public void beforeEnter(BeforeEnterEvent beforeEnterEvent) {
```

Comment:
Why the change of the interface? This part of the code was working fine already, so I would like to know the reason for the change.

Reply:
https://vaadin.com/docs/latest/flow/routing/url-parameters

Reply:
I would appreciate, if functional code done by others is not just overridden but have a discussion first. This increases a) the learning experience and b) shows appreciation.